src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -1337,4 +1339,168 @@ class ActionDelegate {
+     * @param loggedUser the user who is searching for the requests
+     * @param pageable the page configuration that will contain the requests
+     * @param locale the Locale to be used when searching for requests
+     * @param isIntersection to decide null query handling
+     * @return page of cases
+     * */
+    List<Case> findCases(Map<String, Object> request, LoggedUser loggedUser = userService.loggedOrSystem.transformToLoggedUser(),
+                          int pageSize = 25, Locale locale = Locale.default, boolean isIntersection = false) {
+        List<CaseSearchRequest> requests = Collections.singletonList(convertCaseSearchRequest(request))
+        return findCases(requests, loggedUser, pageSize, locale, isIntersection)
+    }
+
+    /**
+     * Action API case search function using Elasticsearch database
+     * @param requests the CaseSearchRequest list
+     * @param loggedUser the user who is searching for the requests
+     * @param pageable the page configuration that will contain the requests
+     * @param locale the Locale to be used when searching for requests
+     * @param isIntersection to decide null query handling
+     * @return page of cases
+     * */
+    List<Task> findTasks(List<ElasticTaskSearchRequest> requests, LoggedUser loggedUser = userService.loggedOrSystem.transformToLoggedUser(),
+                         int pageSize = 25, Locale locale = Locale.default, boolean isIntersection = false) {
+        List<Task> tasks = new ArrayList<>()
+        int numOfPages = (int) ((elasticTaskService.count(requests, loggedUser, locale, isIntersection) / pageSize) + 1)
+        (0..(numOfPages - 1)).each {tasks.addAll(elasticTaskService.search(requests, loggedUser, PageRequest.of(it, pageSize), locale, isIntersection).toList())}
+        return tasks
+    }
+
+    /**
+     * Action API case search function using Elasticsearch database
+     * @param request case search request
+     * @param loggedUser the user who is searching for the requests
+     * @param pageable the page configuration that will contain the requests
+     * @param locale the Locale to be used when searching for requests
+     * @param isIntersection to decide null query handling
+     * @return page of cases
+     * */
+    List<Task> findTasks(Map<String, Object> request, LoggedUser loggedUser = userService.loggedOrSystem.transformToLoggedUser(),
+                         int pageSize = 25, Locale locale = Locale.default, boolean isIntersection = false) {
+        List<ElasticTaskSearchRequest> requests = Collections.singletonList(convertTaskSearchRequest(request))
+        return findTasks(requests, loggedUser, pageSize, locale, isIntersection)
+    }
+
+    /**
+     * Mapper function to create CaseSearchRequest object from map
+     * @param request map of request attributes
+     * @return CaseSerachRequest object
+     * */
+    CaseSearchRequest convertCaseSearchRequest(Map<String, Object> request) {
```

Comment:
this would be better as an alternative constructor / static method of the `CaseSearchRequest` class instead of an action delegate method

Reply:
Done